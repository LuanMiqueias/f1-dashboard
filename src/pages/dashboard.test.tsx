import { render, screen } from "@testing-library/react";
import Dashboard from "./dashboard";

const DriversFake = [
  {
    position: 1,
    driver: {
      id: 1,
      name: "string",
      image: "string",
    },
    team: {
      id: 0,
      name: "string",
      logo: "string",
    },
    points: 0,
    season: 0,
  },
];

const competitionsFake = [
  {
    id: 1,
    name: "string",
    location: {
      country: "string",
      city: "string",
    },
  },
];
it("Should math snapshot", () => {
  const { container } = render(
    <Dashboard
      drivers={DriversFake}
      competitions={competitionsFake}
      error={null}
    />
  );

  expect(container).toMatchSnapshot();
});

it("Should show message if contain error", () => {
  render(
    <Dashboard
      drivers={DriversFake}
      competitions={competitionsFake}
      error={"error"}
    />
  );

  expect(screen.getByText(/error/i)).toBeInTheDocument;
});
