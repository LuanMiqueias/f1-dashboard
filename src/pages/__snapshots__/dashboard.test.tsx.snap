// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should math snapshot 1`] = `
<div>
  <header
    class="container"
  >
    <div
      class="content"
    >
      <img
        alt="logo"
        class="logo"
        src="logo-f1.svg"
      />
      <p
        class="year"
      >
        2021
      </p>
    </div>
  </header>
  <div
    class="container"
  >
    <div
      class="content"
    >
      <div
        class="box"
      >
        <div
          class="content_top"
        >
          <div
            class="title_box"
          >
            <h1>
              Ranking drives
            </h1>
          </div>
        </div>
        <div
          class="container"
        >
          <div
            class="card"
          >
            <p>
              <span
                class="colorPosition"
              />
              1
              Â°
            </p>
            <div
              class="content_card"
            >
              <h2>
                string
              </h2>
              <p>
                string
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="box"
      >
        <div
          class="content_top"
        >
          <div
            class="title_box"
          >
            <h1>
              Races
            </h1>
            <p>
              select one competitions
            </p>
          </div>
        </div>
        <div
          class="container"
        >
          <div
            class="card_competition"
          >
            <div
              class="content_card_competition"
            >
              <h2>
                string
              </h2>
              <p>
                string
                , 
                string
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
